bag_unpacker/src/main/scala/uk/ac/wellcome/platform/archive/bagunpacker/services/S3Uploader.scala

```diff
@@ -8,12 +8,31 @@ import com.amazonaws.services.s3.transfer.model.UploadResult
 import com.amazonaws.services.s3.transfer.{TransferManagerBuilder, Upload}
 import uk.ac.wellcome.storage.ObjectLocation
 
-class S3Uploader(bufferSize: Int)(implicit s3Client: AmazonS3) {
+class S3Uploader(implicit s3Client: AmazonS3) {
 
   private val transferManager = TransferManagerBuilder.standard
     .withS3Client(s3Client)
     .build
 
+  // Set bufferReadLimit
+  // To prevent 'com.amazonaws.ResetException: Failed to reset the request input stream' being thrown.
+  // (see https://github.com/aws/aws-sdk-java/issues/427)
+  //
+  // If a transfer fails, the AWS SDK retries by rewinding the input stream to the 'mark' set in the buffer at the start.
+  // The 'ReadLimit' determines how far the stream can be rewound, if it is smaller than the bytes sent before an error
+  // occurs the mark will be invalid and a ResetException is thrown.
+  //
+  // When using a BufferedInputStream up to 'ReadLimit' bytes are stored in memory, and this must be (at least one byte)
+  // larger than the PUT.  The buffer grows dynamically up to this limit.
+  //
+  // For multipart PUT requests the size must be larger than each PART.
+  //
+  // To prevent this exception a constant maximum size is set
+  // assuming a maximum file of 600GB PUT as 10,000 multipart requests
+  // = 60MB ~ 100MB read limit
```

Comment:
I’d add a link to our ticket as well, so future readers can reconstruct our attempts to fix it.

And possibly a note that 100MB * 10 streams = 1GB, so we're comfortable this will fit into memory?